### PR TITLE
Issue#109: CSVtoJson conversion test done

### PR DESCRIPTION
**Pull request by luhyxi**
This creates a xunit test for CSVtoJson conversion in the CsvParserTests, as requested by Issue#109

- - -
This change is [Reviewable](https://reviewable.io/reviews/cs-util-com/cscore/121)

### CODE REVIEW

File: CsCore/xUnitTests/xUnitTests.csproj [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <TargetFramework>net6.0</TargetFramework>
-	<LangVersion>8.0</LangVersion>
+	<LangVersion>8</LangVersion>
```
Comment (cs-util) [inline CsCore/xUnitTests/xUnitTests.csproj:5]: Can you revert this change since its unrelated to the topic of the PR? You can manually edit the file with a text editor

File: CsCore/xUnitTests/src/Plugins/CsCoreXUnitTests/com/csutil/tests/http/CsvParserTests.cs [hunk 2]
```diff
@@ -55,7 +57,34 @@ private class Person {
...
+            Assert.Equal(3, dataAsJson.Count); // 2 rows
+            Assert.Equal(3, ((JArray)dataAsJson[0]).Count); // 3 columns
+
+            Assert.Equal("Name", dataAsJson[0][0].ToString());
+            Assert.Equal("Age", dataAsJson[0][1].ToString());
+            Assert.Equal("Location", dataAsJson[0][2].ToString());
+
+            Assert.Equal("Alice", dataAsJson[1][0].ToString());
+            Assert.Equal("25", dataAsJson[1][1].ToString());
+            Assert.Equal("New York", dataAsJson[1][2].ToString());
```
Comment (cs-util) [inline CsCore/xUnitTests/src/Plugins/CsCoreXUnitTests/com/csutil/tests/http/CsvParserTests.cs:86]: Looks good so far, now that the data is in a JArray it would be great if you could parse it also to a MyUser class you create here in the test and use the JsonReader.GetReader().Read<List<MyUser>>(dataAsJson) to parse it into a second structure where you do the same assertions on as here above. I would say this would nicely complete the demonstration how to easy work with CSVs and parse them into custom classes. 

If that works nice feel free to refactor the test by extracting this part into its own method and then moving it into the CsvParser class: 
```
            List<List<string>> parsedData = CsvParser.ReadCsvStream(stream);

            JArray dataAsJson = new JArray();

            foreach (var row in parsedData)
            {
                JArray rowArray = new JArray(row.Select(JToken.FromObject));
                dataAsJson.Add(rowArray);
            }
```

**Review by cs-util — CHANGES_REQUESTED** [review]

File: CsCore/xUnitTests/xUnitTests.csproj (hunk 1 above)
Comment (cs-util) [reply CsCore/xUnitTests/xUnitTests.csproj:5]: Or actually never mind, if your IDE did this change automatically lets keep it at this and not revert it, otherwise every other PR other people would create would also contain this change. So please just add a comment here that this was done by your IDE automatically

**Comment by luhyxi** [comment]
Redone the logic and the tests as you said, I put most of the logic inside the CsvParser and made it easier to understand the behavior of the parser.

File: CsCore/PlainNetClassLib/src/Plugins/CsCore/com/csutil/io/CsvParser.cs [hunk 3]
```diff
@@ -28,8 +28,16 @@ public class CsvParser {
...
+            List<List<string>> parsedData = CsvParser.ReadCsvStream(stream);
+            return CsvParser.ConvertToJson(parsedData);
+        }
+
+        public static JArray ConvertCsvToJArray(List<List<string>> parsedCsvData) {
+            return CsvParser.ConvertToJson(parsedCsvData);
+        }
 
-        public static JArray ConvertToJson(List<List<string>> parsedCsvData) {
+        internal static JArray ConvertToJson(List<List<string>> parsedCsvData) {
```
Comment (cs-util) [inline CsCore/PlainNetClassLib/src/Plugins/CsCore/com/csutil/io/CsvParser.cs:40]: To keep cscore backwards compatible for everyone potentially already using this method, here instead of switching it to internal directly, first you have to set it to Obsolete with a note on what new method to use instead, then in eg a few months this method can be switched to internal

File: CsCore/xUnitTests/src/Plugins/CsCoreXUnitTests/com/csutil/tests/http/CsvParserTests.cs [hunk 4]
```diff
@@ -56,6 +57,66 @@ private class Person {
...
+
+            // Creating a MemoryStream from the CSV data
+            using var stream = new MemoryStream(Encoding.Default.GetBytes(sampleCsvData));
+
+            // Converting CSV data to a JSON array using CsvParser
+            var dataAsJson = CsvParser.ReadCsvStreamAsJson(stream);
+            
+
+            // Creating a List of MyUsers
+            List<MyUser> users = new List<MyUser>(dataAsJson.ToObject<List<MyUser>>());
```
Comment (cs-util) [inline CsCore/xUnitTests/src/Plugins/CsCoreXUnitTests/com/csutil/tests/http/CsvParserTests.cs:92]: Lets put these 2 methods together into a 
`List<MyUser> users = CsvParser.ReadCsvStreamAs<MyUser>(stream);`

Inside this `public List<T> ReadCsvStreamAs<T>(stream)` method please use the JsonReader wrapper class so that a developer can decide themselves if they use Newtonsoft for the json parsing (which is the default implementation that JsonReader does) or if they use something else internally.

**Review by cs-util — CHANGES_REQUESTED** [review]
Looks good :) I added a few more comments based on the new changes
